wrappers/services.go

```diff
@@ -376,28 +376,54 @@ func userDaemonReload() error {
 	return cli.ServicesDaemonReload(ctx)
 }
 
+// EnsureSnapServicesOptions is the set of options applying to the
+// EnsureSnapServices operation. It does not include per-snap specific options
+// such as VitalityRank or RequireMountedSnapdSnap from AddSnapServiceOptions,
+// since those are expected to be provided in the snaps argument.
+type EnsureSnapServicesOptions struct {
+	// Preseeding is whether the system is currently being preseeded, in which
+	// case there is not a running systemd for EnsureSnapServicesOptions to
+	// issue commands like systemctl daemon-reload to.
+	Preseeding bool
```

Comment:
RequireMountedSnapdSnap is also a global property that should go here

Reply:
maybe I didn't make this clear in my commit messages/comments here, but I think I disagree with your suggestion about making both Preseeding and RequireMountedSnapdSnap global options since it introduces ambiguity over which option takes precedence and is confusing to reason about. For example, what should this function do if the following happens:

* global EnsureSnapServicesOptions.RequireMountedSnapdSnap is false, per-snap AddSnapServicesOptions.RequireMountedSnapdSnap is true
* global EnsureSnapServicesOptions.RequireMountedSnapdSnap is true, per-snap AddSnapServicesOptions.RequireMountedSnapdSnap is false

I originally did include RequireMountedSnapdSnap here, but then I found myself writing code like this:

```go
processedOpt.RequireMountedSnapdSnap := (perSnapOpt.RequireMountedSnapdSnap || globalOpt.RequireMountedSnapdSnap)
```

in order to make all the unit tests pass. This sort of duplication of options is also confusing as a reader of the function to explain that there are multiple options to provide to the function, and that the net option that goes into effect is an OR of all the options provided. I think it is far simpler to explain that there are some globally applicable options you set in EnsureSnapServicesOptions (like Preseeding) that 100% always take effect using the EnsureSnapServicesOptions (ignoring entirely AddSnapServicesOptions.Preseeding), and all other options are specified in the per-snap options. This is both easier to reason about and simpler to handle in the code.

Furthermore, in the case of VitalityRank and RequireMountedSnapdSnap, these are options that pertain to _what_ we generate, not _how_ we generate. The EnsureSnapServicesOptions to me is an option bundle about _how_ we are generating things (thus including Preseeding there), and the per-snap options in AddSnapServicesOptions pertains to _what_ we are generating. 

Hopefully this all makes sense.

Reply:
I think the least confusing middle term path is to have:

SnapServiceOptions{VitalityRank}
EnsureSnapServicesOptions{Preseeding,RequireMountedSnapdSnap}

and AddSnapServicesOptions as is for now, it would be split internally to reuse EnsureSnapServices.

The most annoying bit of this is that the generateSnapServiceFile will take a SnapServiceOptions and a EnsureSnapServicesOptions separately.

Reply:
Okay, this is done now